mypy/typeanal.py

```diff
@@ -605,7 +605,7 @@ def try_analyze_special_unbound_type(self, t: UnboundType, fullname: str) -> Typ
                     t,
                     code=codes.VALID_TYPE,
                 )
-            else:
+            elif self.options.python_version < (3, 13):
```

Comment:
We should still error in several of the cases this previously triggered, e.g. `def f(x: Final[int]) -> int: ...` (grep the test cases for `Final can be only used as an outermost qualifier` for examples)